knowledge_graphs/query_pagination.py

```
@@ -0,0 +1,382 @@
+"""
+Query pagination utilities for Neo4j knowledge graph queries.
+"""
+import asyncio
+import logging
+from typing import Dict, Any, List, Optional, AsyncIterator, Tuple
```

Ruff (0.12.2)

Import from `collections.abc` instead: `AsyncIterator`

Import from `collections.abc`

(UP035)